langchain/src/callbacks/handlers/log_stream.ts

```diff
@@ -0,0 +1,304 @@
+import jsonpatch from "../../util/fast-json-patch/fast-json-patch.js";
+import { BaseTracer, type Run } from "./tracer.js";
+import { BaseCallbackHandlerInput } from "../base.js";
+import { IterableReadableStream } from "../../util/stream.js";
+
+/**
+ * Interface that represents the structure of a log entry in the
+ * `LogStreamCallbackHandler`.
+ */
+export type LogEntry = {
+  /** ID of the sub-run. */
+  id: string;
+  /** Name of the object being run. */
+  name: string;
+  /** Type of the object being run, eg. prompt, chain, llm, etc. */
+  type: string;
+  /** List of tags for the run. */
+  tags: string[];
+  /** Key-value pairs of metadata for the run. */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  metadata: Record<string, any>;
+  /** ISO-8601 timestamp of when the run started. */
+  start_time: string;
+  /** List of LLM tokens streamed by this run, if applicable. */
+  streamed_output_str: string[];
+  /** Final output of this run. Only available after the run has finished successfully. */
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  final_output?: any;
+  /** ISO-8601 timestamp of when the run ended. Only available after the run has finished. */
+  end_time?: string;
+};
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export type RunState = Record<string, any> & {
```

Comment:
Why Record here too? When would this have additional keys?

Reply:
Oversight, you're right!